File: 0066-Plus-One/lc0066.cc
// C++ solution.

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for (int pos = digits.size()-1; pos >= 0; pos--) {
            if (digits[pos] == 9) {
                digits[pos] = 0;
            } else {
                digits[pos] += 1;
                return digits;
            }
        }
        digits[0] = 1;
        digits.push_back(0);
        return digits;
    }
};

File: 0066-Plus-One/lc0066.py
# Python solution.

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if digits == []:
            return [1]
        else:
            t = digits[-1] + 1
            if t < 10:
                return digits[:-1] + [t]
            else:
                return self.plusOne(digits[:-1]) + [0]
